scipy/spatial/_qhull.pyx

```diff
@@ -2929,6 +2929,20 @@ class HalfspaceIntersection(_QhullUser):
         of halfspaces is also not possible after `close` has been called.
 
         """
+        # We check for non-feasibility of incremental additions
+        # in a manner similar to `qh_sethalfspace`
+        halfspaces = np.atleast_2d(halfspaces)
+        dists = np.dot(halfspaces[:, :self.ndim], self.interior_point) + halfspaces[:, -1]
+        # HalfspaceIntersection uses closed half spaces so
+        # the feasible point also cannot be directly on the boundary
+        viols = dists >= 0
```

Comment:
`HalfspaceIntersection` docs suggest that the half spaces it accepts are "closed" so I think we should indeed reject interior points directly on the boundary like this. I believe @RichardBowenGM and I agree on this.